src/main/java/de/tum/in/www1/artemis/service/TextFeedbackMigration.java

```diff
@@ -0,0 +1,97 @@
+package de.tum.in.www1.artemis.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.annotation.PostConstruct;
+
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.Feedback;
+import de.tum.in.www1.artemis.domain.TextBlock;
+import de.tum.in.www1.artemis.domain.TextExercise;
+import de.tum.in.www1.artemis.domain.TextSubmission;
+import de.tum.in.www1.artemis.repository.FeedbackRepository;
+import de.tum.in.www1.artemis.repository.TextBlockRepository;
+import de.tum.in.www1.artemis.repository.TextExerciseRepository;
+
+@Service
+public class TextFeedbackMigration {
+
+    private final Logger log = LoggerFactory.getLogger(TextFeedbackMigration.class);
+
+    private final String textBlockIdRegex = "^[a-f0-9]{40}$";
+
+    private final Pattern textBlockIdPattern = Pattern.compile(textBlockIdRegex);
+
+    private final FeedbackRepository feedbackRepository;
+
+    private final TextExerciseRepository textExerciseRepository;
+
+    private final TextBlockRepository textBlockRepository;
+
+    public TextFeedbackMigration(FeedbackRepository feedbackRepository, TextExerciseRepository textExerciseRepository, TextBlockRepository textBlockRepository) {
+        this.feedbackRepository = feedbackRepository;
+        this.textExerciseRepository = textExerciseRepository;
+        this.textBlockRepository = textBlockRepository;
+    }
+
+    @NotNull
+    private List<TextExercise> findAllTextExercises() {
+        return textExerciseRepository.findAll();
+    }
+
+    @NotNull
+    private List<Long> findAllTextExerciseIds() {
+        return findAllTextExercises().parallelStream().map(TextExercise::getId).collect(Collectors.toList());
+    }
+
+    @NotNull
+    private List<Feedback> findAllTextFeedbacks() {
+        return feedbackRepository.findAllByResult_Submission_Participation_ExerciseIdIn(findAllTextExerciseIds());
+    }
+
+    @NotNull
+    private List<Feedback> findAllTextFeedbacksNeedingMigration() {
+        return findAllTextFeedbacks().parallelStream().filter(feedback -> feedback.getReference() != null && !textBlockIdPattern.matcher(feedback.getReference()).matches())
+                .collect(Collectors.toList());
+    }
+
+    @PostConstruct()
```

Comment:
Build fails here for missing JavaDoc. See comment in [PR description](https://github.com/ls1intum/Artemis/pull/729#issue-308667539):
> So far, this PR does not contain JavaDoc comments. However, these changes will be reverted as soon as it was run on the production instance, so we can save those efforts towards more useful documentation efforts like #717.